### PR TITLE
Feature/create login otp email controller

### PR DESCRIPTION
**Pull request by ismetkizgin**

### CODE REVIEW

**Review by katafalque — APPROVED** [review]

File: src/login/login-otp-email.controller.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    private readonly otpService: OtpService,
+    @Inject("TokenService") private readonly tokenService: TokenService,
+    @Inject("UserService") private readonly userService: UserService
+  ) {}
+
+  @Post("login-otp-email")
+  async loginAsync(
+    @Body() loginOtpEmailRequest: LoginOtpEmailRequest
+  ): Promise<LoginResponse> {
+    const isItConfirmed = await this.otpService.validateEmailOtpAsync(
```
Comment (erailea) [inline src/login/login-otp-email.controller.ts:30]: isConfirmed, isOtpConfirmed seems better naming

File: src/login/login-otp-email.controller.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    );
+
+    if (!isOtpValid) throw new UnauthorizedException("Invalid credentials");
+
+    const user = await this.userService.getUserAsync({
+      email: loginOtpEmailRequest.email,
+    });
+
+
+    return await this.tokenService.createTokensAsync(user);
```
Comment (mfozmen) [inline src/login/login-otp-email.controller.ts:42]: Check if the user exists.

File: test/integration/login-otp-email.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  });
+
+  afterEach(async () => {
+    await app.close();
+    await moduleFixture.close();
+  });
+
+  describe("POST /login-otp-email", () => {
+    it("Should return a token if OTP credentials are valid", async () => {
+      const mockUser = MockFactory(UserFixture).one().hashPassword();
```
Comment (mfozmen) [inline test/integration/login-otp-email.e2e-spec.ts:43]: General note for namings. Do not name as mockUser, mockOtp etc. Define the variable well. Which user is it? Which otp is it? Name it well for better understanding the test.

File: test/integration/login-otp-email.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    otpRepository = moduleFixture.get<Repository<Otp>>("OtpRepository");
+  });
+
+  afterEach(async () => {
+    await app.close();
+    await moduleFixture.close();
+  });
+
+  describe("POST /login-otp-email", () => {
+    it("Should return a token if OTP credentials are valid", async () => {
```
Comment (mfozmen) [inline test/integration/login-otp-email.e2e-spec.ts:42]: should return tokens if otp is valid

Comment (mfozmen) [inline test/integration/login-otp-email.e2e-spec.ts:42]: Add user not found case test also.

**Review by mfozmen — CHANGES_REQUESTED** [review]

**Review by mfozmen — APPROVED** [review]

**Review by mfozmen — APPROVED** [review]

**Pull request merged**